### ISSUE TITLE
PASTA Fine-tuning for Esophageal Cancer Segmentation: Large Discrepancy between Pseudo/EMA Dice + Abnormal Predictions

### ISSUE THREAD

**Issue by yangboyan2026** [LWHYC/PASTA#8]
Dear authors, thank you for your excellent work!

I used PASTA to fine-tune an esophageal cancer segmentation model (120 real data samples) with the following training command:
```bash
python segmentation/nnunetv2/run/run_finetuning_pasta.py 501 3d_fullres 0 -tr PASTATrainer -pretrained_weights /home/PASTA_final.pth
```

I encountered two key problems:
1. The Pseudo Dice score is very high, but the EMA pseudo Dice score is extremely low, and the final validation Dice score is only 0.4;
2. Predictions often result in Dice=0 (complete failure to segment), and even for results with Dice > 0.85, ITK-SNAP shows a large number of incorrect ROIs.
3. Has the training of PASTA_final.pth incorporated all the contents of the LWHYC/PASTA-Gen-30K dataset?

I would like to ask for advice on how to optimize these two issues. Thank you!

**Comment by LWHYC** [comment]
> Dear authors, thank you for your excellent work!
> 
> I used PASTA to fine-tune an esophageal cancer segmentation model (120 real data samples) with the following training command:
> 
> python segmentation/nnunetv2/run/run_finetuning_pasta.py 501 3d_fullres 0 -tr PASTATrainer -pretrained_weights /home/PASTA_final.pth
> I encountered two key problems:
> 
> 1. The Pseudo Dice score is very high, but the EMA pseudo Dice score is extremely low, and the final validation Dice score is only 0.4;
> 2. Predictions often result in Dice=0 (complete failure to segment), and even for results with Dice > 0.85, ITK-SNAP shows a large number of incorrect ROIs.
> 3. Has the training of PASTA_final.pth incorporated all the contents of the LWHYC/PASTA-Gen-30K dataset?
> 
> I would like to ask for advice on how to optimize these two issues. Thank you!

Hi,

Thank you for your interest and for trying out PASTA on your task!

First, we would like to apologize for the confusion in the previous guideline. There was an issue in the documentation:
for finetuning from pretrained weights, you should use PASTATrainer_ft instead of PASTATrainer.

The PASTATrainer is designed for the pretraining stage, and its learning rate is relatively high, which can lead to unstable finetuning behavior such as the issues you observed (e.g., mismatch between pseudo Dice and EMA Dice, or failed segmentations).

Regarding your second question: yes, the training of PASTA_final.pth was performed using the full PASTA-Gen-30K dataset.

We recommend switching to PASTATrainer_ft for finetuning and re-running the experiment. This should significantly improve stability and performance.

Thanks again for your valuable feedback!

Best regards
